staging/src/k8s.io/apiextensions-apiserver/test/integration/helpers.go

```diff
@@ -188,3 +194,55 @@ func hasPerVersionColumns(versions []apiextensionsv1beta1.CustomResourceDefiniti
 	}
 	return false
 }
+
+type etcdObjectReader struct {
```

Comment:
we might even want this in a separate `storage` package. What do you think?

Reply:
Sounds good. I'll give it a shot.